peers2/peer_pool.go

```diff
@@ -0,0 +1,259 @@
+package peers2
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/p2p/discover"
+	"github.com/ethereum/go-ethereum/p2p/discv5"
+	"github.com/status-im/status-go/peers"
+)
+
+var (
+	ErrPeerPoolStarted    = errors.New("PeerPool is already started")
+	ErrPeerPoolNotStarted = errors.New("PeerPool is not started")
+)
+
+type p2pServer interface {
+	SubscribeEvents(chan *p2p.PeerEvent) event.Subscription
+	Self() *discover.Node
+	PeersInfo() []*p2p.PeerInfo
+	AddPeer(*discover.Node)
+	RemovePeer(*discover.Node)
+}
+
+type addPeerReq struct {
+	t    discv5.Topic
+	node *discv5.Node
+}
+
+type peerInfo struct {
+	topics    []discv5.Topic
+	node      *discv5.Node
+	dismissed bool
+}
+
+type PeerPool struct {
+	sync.RWMutex
+
+	topics           []TopicPool
+	topicToTopicPool map[discv5.Topic]TopicPool
```

Comment:
maybe just `topicPools` (plural)? I struggled for a minute to understand what `Topic-to-topic Pool` is :)